crates/wasmparser/src/validator/types.rs

```diff
@@ -480,17 +482,17 @@ pub enum InterfaceType {
     /// The type is a primitive interface type.
     Primitive(PrimitiveInterfaceType),
     /// The type is a record.
-    Record(HashMap<String, InterfaceTypeRef>),
+    Record(IndexMap<String, InterfaceTypeRef>),
     /// The type is a variant.
-    Variant(HashMap<String, VariantCase>),
+    Variant(IndexMap<String, VariantCase>),
```

Comment:
Is there a reason not to use `BTreeMap` here? Do we actually need to sort by insertion order here, or does ordering by key work?

Reply:
I'm updating `wit-component` to directly use the type information from validation (this PR) and it's creating `wit_parser::Interface`s directly from that type information; so it needs a way to ensure that the declaration order in the original encoding is also respected in any wit variants it generates.